open-api/rest-catalog-open-api.yaml

```diff
@@ -2068,6 +2162,145 @@ components:
           items:
             $ref: '#/components/schemas/PartitionStatisticsFile'
 
+    PlanTask:
+      description:
+        A JSON object that contains information provided by the server,
+        to be utilized by clients for distributed planning, should be supplied
+        as is for input in PlanTable operation.
+      type: object
+
+    FileScanTask:
+      type: object
+      required:
+        - schema
+        - spec
+        - start
+        - length
+        - data-file
+      properties:
+        data-file:
+          $ref: '#/components/schemas/ContentFile'
+        partition:
+          type: object
+          additionalProperties:
+            type: string
+        size-bytes:
+          type: number
+        start:
+          type: number
+        length:
+          type: number
+        estimated-rows-count:
+          type: number
+        delete-files:
+          type: array
+          items:
+            $ref: '#/components/schemas/ContentFile'
+        schema:
+          $ref: '#/components/schemas/Schema'
+        spec:
+          $ref: '#/components/schemas/PartitionSpec'
+        residual-filter:
+          $ref: '#/components/schemas/Expression'
+
+    TypeValue:
+      oneOf:
+        - $ref: '#/components/schemas/PrimitiveTypeValue'
+        - $ref: '#/components/schemas/MapTypeValue'
+        - $ref: '#/components/schemas/StructTypeValue'
+
+    MapTypeValue:
+      type: object
+      properties:
+        keys:
+          type: array
+          items:
+            $ref: '#/components/schemas/TypeValue'
+        values:
+          type: array
+          items:
+            $ref: '#/components/schemas/TypeValue'
+
+    StructTypeValue:
+      type: object
+      additionalProperties:
+        oneOf:
+          - $ref: '#/components/schemas/TypeValue'
+
+    PrimitiveTypeValue:
+      oneOf:
+        - type: boolean
+        - type: integer
+        - type: integer
+          format: int64
+        - type: number
+        - type: string
+        - type: string
+          format: byte
+        - type: array
```

Comment:
I don't think it makes sense for this to be part of `PrimitiveTypeValue`. I also can't think of a situation in which this is going to be used so I think you can remove it.

Reply:
This is reviewed in more details in https://github.com/apache/iceberg/pull/9717, you might want to jump there. This is pending the change there to update this part.

Reply:
have rebased on pr 9717 https://github.com/apache/iceberg/commit/bb53c3d4e0e27ac6706803c2371793ad2476ae04